src/preview.cpp

```diff
@@ -803,13 +803,13 @@ namespace {
                 else {
                     // FIXME: the buffer is probably copied twice, it should be optimized
                     DataBuf buf(size_);
-                    Exiv2::byte* pos = buf.pData_;
+                    uint32_t idxBuf = 0;
                     for (int i = 0; i < sizes.count(); i++) {
                         uint32_t offset = dataValue.toLong(i);
                         uint32_t size = sizes.toLong(i);
                         if (size!=0 && offset + size <= static_cast<uint32_t>(io.size()))
-                            memcpy(pos, base + offset, size);
-                        pos += size;
+                            memcpy(reinterpret_cast<byte *>(buf.pData_[idxBuf]), base + offset, size);
```

Comment:
Uhm, are you sure about this one? `buf.pData_[idxBuf]` is a `byte`, not `byte *`. Wont this wreak havoc as this code now tries to copy into the memory address stored in `buf.pData_[idxBuf]`?